packages/webview_flutter/webview_flutter_android/lib/src/android_webview_controller.dart

```diff
@@ -102,18 +117,61 @@ class AndroidWebViewController extends PlatformWebViewController {
         }
       };
     }),
-    onShowFileChooser: withWeakReferenceTo(this,
-        (WeakReference<AndroidWebViewController> weakReference) {
-      return (android_webview.WebView webView,
-          android_webview.FileChooserParams params) async {
-        if (weakReference.target?._onShowFileSelectorCallback != null) {
-          return weakReference.target!._onShowFileSelectorCallback!(
-            FileSelectorParams._fromFileChooserParams(params),
-          );
-        }
-        return <String>[];
-      };
-    }),
+    onShowFileChooser: withWeakReferenceTo(
+      this,
+      (WeakReference<AndroidWebViewController> weakReference) {
+        return (android_webview.WebView webView,
+            android_webview.FileChooserParams params) async {
+          if (weakReference.target?._onShowFileSelectorCallback != null) {
+            return weakReference.target!._onShowFileSelectorCallback!(
+              FileSelectorParams._fromFileChooserParams(params),
+            );
+          }
+          return <String>[];
+        };
+      },
+    ),
+    onPermissionRequest: withWeakReferenceTo(
+      this,
+      (WeakReference<AndroidWebViewController> weakReference) {
+        return (_, android_webview.PermissionRequest request) async {
+          final void Function(PlatformWebViewPermissionRequest)? callback =
+              weakReference.target?._onPermissionRequestCallback;
+          if (callback == null) {
+            return request.deny();
+          } else {
+            final List<WebViewPermissionResourceType> types = request.resources
+                .map<WebViewPermissionResourceType?>((String type) {
+                  switch (type) {
+                    case android_webview.PermissionRequest.videoCapture:
+                      return WebViewPermissionResourceType.camera;
+                    case android_webview.PermissionRequest.audioCapture:
+                      return WebViewPermissionResourceType.microphone;
+                    case android_webview.PermissionRequest.midiSysex:
+                      return AndroidWebViewPermissionResourceType.midiSysex;
+                    case android_webview.PermissionRequest.protectedMediaId:
+                      return AndroidWebViewPermissionResourceType
+                          .protectedMediaId;
+                  }
+
+                  // Type not supported.
+                  return null;
```

Comment:
Hm, I'm worried about a scenario where there were two unknown permissions. And then only one was granted on Android. And if we changed to a `Set` of permissions instead of a `List`, this would also not be clear.

Reply:
Ah, that's true. I guess we should just make sure we gracefully filter them out (e.g., not calling the callback if the request set ends up empty), and then rely on people filing feature requests to add support for new types.

Reply:
Added a test to verify the callback is not made when an unknown type is passed.